currency-rate/internal/mail/transport/producer.go

```diff
@@ -0,0 +1,81 @@
+package transport
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+
+	"github.com/GenesisEducationKyiv/software-engineering-school-4-0-Hukyl/currency-rate/internal/mail/transport/config"
+	amqp "github.com/rabbitmq/amqp091-go"
+)
+
+func logAndWrap(msg string, err error) error {
+	slog.Error(msg, slog.Any("error", err))
+	return fmt.Errorf("%s: %w", msg, err)
+}
+
+type Producer struct {
+	config  config.Config
+	conn    *amqp.Connection
+	channel *amqp.Channel
+}
+
+func (p *Producer) Close() error {
+	if err := p.channel.Close(); err != nil {
+		return logAndWrap("closing channel", err)
+	}
+	if err := p.conn.Close(); err != nil {
+		return logAndWrap("closing connection", err)
+	}
+	return nil
+}
+
+func (p *Producer) Produce(ctx context.Context, msg []byte) error {
+	err := p.channel.PublishWithContext(ctx,
+		"",                 // exchange
+		p.config.QueueName, // routing key
+		false,              // mandatory
+		false,              // immediate
+		amqp.Publishing{
+			ContentType: "text/plain",
```

Comment:
if I got it correctly, you're sending messages in JSON format, so are you sure this content type is correct?